packages/transformers/stylus/src/StylusTransformer.js

```diff
@@ -225,21 +269,79 @@ async function createEvaluator(code, asset, resolve, options, parcelOptions) {
             return mergeBlocks(
               resolved.map(resolvedPath => {
                 node.string = resolvedPath;
-                return super.visitImport(imported.clone());
+                let restore = patchNativeFS(asset.fs, nativeGlob);
+                let res = super.visitImport(imported.clone());
+                restore();
+                return res;
               }),
             );
           }
         }
       }
 
+      // Patch the native FS so stylus uses Parcel's FS to read the file.
+      let restore = patchNativeFS(asset.fs, nativeGlob);
+
       // Done. Let stylus do its thing.
-      return super.visitImport(imported);
+      let res = super.visitImport(imported);
+
+      restore();
+      return res;
     }
   }
 
   return CustomEvaluator;
 }
 
+function patchNativeFS(fs, nativeGlob) {
```

Comment:
Could it happen that the the places where `patchNativeFS` is used are interleaved, so that this happens:?
```js
r1 = patchNativeFS()
r2 = patchNativeFS()
r2()
// fs calls here aren't tracked
r1()
```

Reply:
I think that should still work? Restoring the second should restore the fs functions back to the first, not the original.